Add group-by logic

mebster opened the pull request.

mebster committed:
Add group-by logic

konstan committed:
pep8 and some other minor stuff

konstan commented:
There is an issue in generated .csv file.  The number of generated columns may exceed the number of declared columns in the header (when there are multiple clouds per user or multiple users per cloud).  Here is on the example of group by user:

<img width="620" alt="screen shot 2017-07-26 at 08 16 42" src="https://user-images.githubusercontent.com/353767/28607224-25cf9a44-71db-11e7-93e4-bb6a53e84ddd.png">

The list of items in the last column should not be coma-separated.

```
$ cat metrics--2017-06--by-user.csv
User, VM (hours), CPU (hours), RAM (GB-hours), Disk (GB-hours), Included Clouds
konstan, 178.216666667, 367.736666667, 1321.26708333, 32.3300466667, open-telekom-de1, exoscale-ch-gva, eo-cesnet-cz1
$
```

mebster committed:
Dynamically extract users
including, organization check and process for each organization
now depends on SlipStream API

mebster committed:
Merge branch 'group-by' of github.com:mebster/slipstream-usage-tooling into group-by

# Conflicts:
#	usage.py

mebster committed:
Use ';' as user/cloud separator instead of ','

mebster committed:
Add blacklist and tolarate empty lists with warning

cjdcordeiro commented:
note that script will fail on year transition
